### PR TITLE
Add mosaicjson endpoints

### PR DESCRIPTION
**Pull request by hrodmn**
- Add the `/mosaics` endpoints from impact-tiler to the custom titiler-pgstac runtime
- The structure of the custom runtime got more complex so I re-arranged the code to match [eoapi-devseed](https://github.com/developmentseed/eoapi-devseed)
- Add a settings module for the mosaic backend config

After a few false starts I was able to deploy this branch to the `test` env. The `/mosaics` endpoints work great! I tested out the [visualizing mosaicjson notebook](https://github.com/MAAP-Project/maap-documentation/blob/develop/docs/source/technical_tutorials/visualization/visualizing_mosaicjson.ipynb) with the `titiler_endpoint` set to `https://titiler-pgstac.dit.maap-project.org` and the whole thing worked.

### CODE REVIEW

File: cdk/handlers/raster_handler.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+secret = get_secret_dict(pgstac_secret_arn)
+os.environ.update(
+    {
+        "postgres_host": secret["host"],
+        "postgres_dbname": secret["dbname"],
+        "postgres_user": secret["username"],
+        "postgres_pass": secret["password"],
+        "postgres_port": str(secret["port"]),
+    }
+)
```
Comment (hrodmn) [inline cdk/handlers/raster_handler.py:21]: We could improve performance of the whole application if we set these environment variables on the Lambda itself. I don't know how slow it is to read the secret values, but it is something we have to do on every application startup. That's what I have seen in [`eoapi-devseed`](https://github.com/developmentseed/eoapi-devseed/blob/main/infrastructure/app.py#L135-L149), not sure if there would be a security concern with putting the db credentials as environment variables in the Lambda.

Comment (jjfrench) [reply cdk/handlers/raster_handler.py:21]: Makes sense, security best practices prefers secrets manager or something equivalent but we're not granting external accounts access to view our lambda configurations, so this seems fine

File: cdk/runtimes/eoapi/raster/pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,61 @@
+[project]
+name = "eoapi.raster"
+description = "Custom raster tiling service for MAAP"
+readme = "README.md"
+requires-python = ">=3.8"
```
Comment (jjfrench) [inline cdk/runtimes/eoapi/raster/pyproject.toml:5]: Bump to 3.9 or add classifier for 3.8?

File: cdk/runtimes/eoapi/raster/eoapi/raster/models.py [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+    asset_name: Optional[str] = None
+    name: Optional[str] = None
+    description: Optional[str] = None
+    attribution: Optional[str] = None
+    version: Optional[str] = None
+
+    # override default Search field for collections, which is List[str]
+    collections: Optional[List[str]] = None
+    # overriding limit so we can tell if it's defined or not
+    limit: Optional[int]
```
Comment (jjfrench) [inline cdk/runtimes/eoapi/raster/eoapi/raster/models.py:55]: Should this optional value have a default?

**Review by jjfrench — APPROVED** [review]
Looks good 👍🏼  Thanks for the changes!

File: cdk/handlers/raster_handler.py (hunk 1 above)
Comment (hrodmn) [reply cdk/handlers/raster_handler.py:21]: I will file an issue and we can decide if it's worth changing later.

**Pull request merged**